### PR TITLE
Enable Interpolate mkldnn kernel

### PR DESCRIPTION
**Pull request by lidanqing-vv**
### PR types
Performance optimization

### PR changes
OPs 

### Describe
Enable interpolate mkldnn kernel, so that ocr_det model inference performance is improved, 
Machine: 6148 
Reproduction: #28554 
Performance: 
- CPU Native 7576 ms
- oneDNN 3700 ms

Status: The draft PR open for verifying performance and accuracy. I am polishing up the code.

### CODE REVIEW

**Comment by lidanqing-vv** [comment]
@juncaipeng  The accuracy error should be fixed, I checked first 500 numbers, the same. I will finish all UT tomorrow.

**Comment by juncaipeng** [comment]
> @juncaipeng The accuracy error should be fixed, I checked first 500 numbers, the same. I will finish all UT tomorrow.

OK, The accuracy and performance of ocr crnn model

**Comment by lidanqing-vv** [comment]
@juncaipeng Since oneDNN use different formula for interpolate. [oneDNN resampling](https://docs.oneapi.com/versions/latest/onednn/dev_guide_resampling.html)
The user may get different result when running oneDNN interpolate compared with Native interpolate, but sometimes users can get same results. But I checked tensorflow, tensorflow used same formula as oneDNN. So we have a question if it is ok that applying oneDNN interpolate sometimes have different results than Native, sometimes the same.

Since other frameworks like Tensorflow is using same formula as oneDNN, the oneDNN accuracy may **slightly** differ compared to Natvie.

**Comment by juncaipeng** [comment]
@lidanqing-intel 
Compared to oneDNN, the interpolate of  Paddle has align_corners parameter and the calculation method is different.
We need to check whether the slight difference affects the final result. For the test image, the ocr detection model has  right result, but we don't test the final accuracy.

**Comment by lidanqing-vv** [comment]
Hi, @jczaja  could you please review this PR

**Comment by lidanqing-vv** [comment]
@juncaipeng  https://github.com/oneapi-src/oneDNN/issues/653

**Comment by luotao1** [comment]
@lidanqing-intel @jczaja Discussed internal, our interpolate API behaves in the same way as Pytorch, not TF. See [torch.nn.functional.interpolate](https://pytorch.org/docs/stable/nn.functional.html?highlight=interpolate#torch.nn.functional.interpolate) and its [chinese explanation](https://www.cnblogs.com/wanghui-garcia/p/11399034.html). Thus, unfortunately, we may not call the oneDNN interpolate kernel.

**Comment by lidanqing-vv** [comment]
@jczaja @wojtuss  Please review

File: paddle/fluid/operators/mkldnn/interpolate_mkldnn_op.cc [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      this->fwd_pd_.reset(new dnnl::resampling_forward::primitive_desc(
+          resampling_d, this->engine_));
+
+      auto key_pd = this->key_ + "@fwd_pd";
+      this->dev_ctx_.SetBlob(key_pd, this->fwd_pd_);
+    }
+  }
+
+  std::shared_ptr<resampling_forward::primitive_desc>
+  AcquireForwardPrimitiveDescriptor() {
```
Comment (jczaja) [inline paddle/fluid/operators/mkldnn/interpolate_mkldnn_op.cc]: 1) Why do you need to define your own AcquireForwardPrimitiveDescriptor. There is generic implementation in MKLDNNHandlerT . Why not use this base one?

2) Currently for primitivedescriptor no info on threads should be added to the key. you used key_ which (for Parallel executor, and dygraph) will contain TID which makes crash during training

File: paddle/fluid/operators/mkldnn/interpolate_mkldnn_op.cc [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+    auto out_dims_vec = ComputeOutputShape(ctx);
+    framework::DDim dim_out = framework::make_ddim(out_dims_vec);
+    z->mutable_data<T>(dim_out, ctx.GetPlace());
+
+    InterpolateMKLDNNHandler<T> handler(algo, dev_ctx, mkldnn_engine,
+                                        ctx.GetPlace(), x, z,
+                                        ctx.OutputName("Out"));
+
+    auto resampling_pd = handler.AcquireForwardPrimitiveDescriptor();
```
Comment (jczaja) [inline paddle/fluid/operators/mkldnn/interpolate_mkldnn_op.cc]: This is not needed here and is set (fwd_pd_) when MKLDNNHandlerT::isCached() is called.

File: paddle/fluid/operators/mkldnn/interpolate_mkldnn_op.cc (hunk 1 above)
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/interpolate_mkldnn_op.cc]: I called AcquireForwardPrimitiveDescriptor in MKLDNNHandlerT now
